src/helpers/helpersAPI.js

```diff
@@ -0,0 +1,20 @@
+const getLocationsFromResponseAPI = (response = null) => {
+  if (response === null) return [];
+  if (typeof response.data === 'undefined') return [];
+  if (typeof response.data.query === 'undefined') return [];
+
+  const { query } = response.data;
+  const pages = typeof query.pages === 'undefined' ? [] : query.pages;
+
+  return typeof pages === 'undefined' ? [] : pages;
+};
+
+const sortLocationsByDistance = (locations = []) => {
```

Comment:
This function would be a good candidate for unit testing. Why?

1. It takes a discrete input and returns the same output each time it’s called with the same input
2. It doesn’t have [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science))
3. There are some [subtleties](https://andrew.hedges.name/blog/numerical-array-sorting-in-javascript) to JavaScript’s `sort` function that it would be good to be aware of, if not fix

Like with the other function in this module, I’m not suggesting we put tests around this right now, but maybe we put a story on the board to add some tests for this function to better understand:

- What does it do when 1 or both values are not numbers?
- What does it do when 1 or both values are `Infinity`?
- What does it do when all of the values are exactly the same?

None of the above are _likely_ to happen, but it’s always a good idea to fully understand the behavior of your code. Sometimes things like this can be subtle security holes, for example!

Reply:
Wow, it is incredible 😮!. 

You gave us great info about **_unit testing_**! Excellent explanation. Thank you so much!

Also, I believe your articles need to be shared with more frequency because they are very helpful. Congrats! Also, I learned that `a-b` is not a good way to sort an array.